### ISSUE TITLE
Make configurations with inline input and output nodes invalid.

### ISSUE THREAD

**Issue by jetuk** [pywr/pywr#913]
In my opinion the main solvers (glpk and lpsolve) should not accept networks with inline input and output nodes. I think this configuration should be checked and made invalid in those solvers, and perhaps also made impossible to create via the Python API (i.e. `.connect`).

My argument for this being a bug is that the treatment of the constraints on inline inputs and outputs is misleading. It appears from the network as though these constraints would be somehow imposed on routes passing through them. In fact they are not, and nor can they be made to in an intuitive way. Therefore, I think we should just make configuring models like this invalid. 

This *will* break some existing models.

@Batch21 thoughts?

**Comment by Batch21** [comment]
yeah, I agree. It will be much cleaner and easier to make them invalid. It shouldn't be too difficult to update models that this change breaks.
